google/cloud/firestore_v1/base_document.py

```diff
@@ -243,8 +243,11 @@ def __eq__(self, other):
         return self._reference == other._reference and self._data == other._data
 
     def __hash__(self):
-        seconds = self.update_time.seconds
-        nanos = self.update_time.nanos
+        # TODO(microgen): maybe add datetime_with_nanos to protoplus, revisit
+        # seconds = self.update_time.seconds
+        # nanos = self.update_time.nanos
+        seconds = int(self.update_time.timestamp())
```

Comment:
my concern with floats is minor differences could result in a hash miss when there isn't. It may be good to make an issue here though. I want to add DatetimeWithNanos to protoplus so we can use that.
https://github.com/googleapis/proto-plus-python/issues/38